### PR TITLE
feat: add native Xdebug support

### PR DESCRIPTION
**Pull request by dunglas**
| Q             | A
| ------------- | ---
| Branch?       | main
| Tickets       | n/a
| License       | MIT
| Doc PR        | todo

See https://github.com/dunglas/symfony-docker/pull/271

### CODE REVIEW

File: api/docker/php/conf.d/api-platform.dev.ini [hunk 1]
```diff
@@ -1,13 +1,5 @@
...
-realpath_cache_ttl = 600
-opcache.interned_strings_buffer = 16
-opcache.max_accelerated_files = 20000
-opcache.memory_consumption = 256
-opcache.enable_file_override = 1
+; See https://docs.docker.com/desktop/networking/#i-want-to-connect-from-a-container-to-a-service-on-the-host
+; See https://github.com/docker/for-linux/issues/264
+; The `client_host` below may optionally be replaced with `discover_client_host=yes`
+; Add `start_with_request=yes` to start debug session on each request
+xdebug.client_host = 'host.docker.internal'
```
Comment (vincentchalamon) [inline api/docker/php/conf.d/api-platform.dev.ini:5]:
```suggestion
zend_extension = xdebug.so
xdebug.start_with_request = yes
xdebug.client_host = 'host.docker.internal'
```

**Review by vincentchalamon — CHANGES_REQUESTED** [review]
We could also add the `launch.json` file for VSCode:
```json
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Listen for Xdebug",
            "type": "php",
            "request": "launch",
            "pathMappings": {
              "/srv/api": "${workspaceRoot}/api"
            }
        }
    ]
}
```
(file to test, I'm not 100% sure about its location and if it misses a `port` option)

File: api/docker/php/conf.d/api-platform.dev.ini (hunk 1 above)
Comment (dunglas) [reply api/docker/php/conf.d/api-platform.dev.ini:5]: Adding the xdebug extension is done automatically, and I don't think that's a good idea to profile every requests by default.

**Comment by dunglas** [comment]
To not bloat this skeleton and keep it as minimal as possible, I would prefer having the VSCode configuration in the docs instead.

**Review by vincentchalamon — APPROVED** [review]

**Review by maxhelias — APPROVED** [review]

File: api/docker/php/docker-entrypoint.sh [hunk 2]
```diff
@@ -7,11 +7,7 @@ if [ "${1#-}" != "$1" ]; then
 fi
 
 if [ "$1" = 'php-fpm' ] || [ "$1" = 'php' ] || [ "$1" = 'bin/console' ]; then
-	if [ "$APP_ENV" != 'prod' ]; then
-		ln -sf "$PHP_INI_DIR/php.ini-development" "$PHP_INI_DIR/php.ini"
-	fi
-
-	mkdir -p var/cache var/log
+	#mkdir -p var/cache var/log
```
Comment (vincentchalamon) [inline api/docker/php/docker-entrypoint.sh:10]:
```suggestion
```

File: pwa/Dockerfile [hunk 3]
```diff
@@ -18,13 +18,17 @@ ENV NEXT_TELEMETRY_DISABLED 1
 COPY package.json yarn.lock ./
 RUN yarn
 
-COPY . ./
+COPY . .
+
+VOLUME /usr/src/pwa/node_modules
```
Comment (vincentchalamon) [inline pwa/Dockerfile:23]: Isn't it breaking?

Comment (dunglas) [reply pwa/Dockerfile:23]: No, it's removing this: https://github.com/api-platform/api-platform/commit/92b6f245d9b6a415fa9c4ad58dac084e8a2f35a0

**Review by vincentchalamon — APPROVED** [review]

**Pull request merged**